frontend/appflowy_flutter/integration_test/desktop/grid/grid_reorder_row_test.dart

```diff
@@ -0,0 +1,209 @@
+
+      // reorder row
+      await tester.reorderRow(original[4], original[1]);
+
+      // verify grid data
+      List reordered = [
+        original[0],
+        original[4],
+        original[1],
+        original[2],
+        original[3],
+        original[5],
+        original[6],
+        original[7],
+        original[8],
+        original[9],
+        original[10],
+        original[11],
+        original[12],
+      ];
+      List actual = tester.getGridRows();
+      expect(actual, orderedEquals(reordered));
+
+      // reorder row
+      await tester.reorderRow(reordered[1], reordered[3]);
+
+      // verify grid data
+      reordered = [
+        original[0],
+        original[1],
+        original[4],
+        original[2],
+        original[3],
+        original[5],
+        original[6],
+        original[7],
+        original[8],
+        original[9],
+        original[10],
+        original[11],
+        original[12],
+      ];
+      actual = tester.getGridRows();
+      expect(actual, orderedEquals(reordered));
+
+      // reorder row
+      await tester.reorderRow(reordered[2], reordered[0]);
+
+      // verify grid data
+      reordered = [
+        original[4],
+        original[0],
+        original[1],
+        original[2],
+        original[3],
+        original[5],
+        original[6],
+        original[7],
+        original[8],
+        original[9],
+        original[10],
+        original[11],
+        original[12],
+      ];
+      actual = tester.getGridRows();
+      expect(actual, orderedEquals(reordered));
+    });
+
+    testWidgets('with active sort', (tester) async {
+      await tester.openTestDatabase(v069GridFileName);
+
+      // get grid data
+      final original = tester.getGridRows();
+
+      // add a sort
+      await tester.tapDatabaseSortButton();
+      await tester.tapCreateSortByFieldType(FieldType.RichText, 'Name');
+
+      // verify grid data
+      final sorted = [
+        original[7],
+        original[8],
+        original[1],
+        original[9],
+        original[11],
+        original[10],
+        original[6],
+        original[12],
+        original[2],
+        original[0],
+        original[3],
+        original[5],
+        original[4],
+      ];
+      List actual = tester.getGridRows();
+      expect(actual, orderedEquals(sorted));
+
+      // reorder row
+      await tester.reorderRow(original[4], original[1]);
+
+      // verify grid data
+      actual = tester.getGridRows();
+      // TODO(RS): remind users why the reorder failed
+      expect(actual, orderedEquals(sorted));
+    });
+
+    testWidgets('with active filter', (tester) async {
+      await tester.openTestDatabase(v069GridFileName);
+
+      // get grid data
+      final original = tester.getGridRows();
+
+      // add a filter
+      await tester.tapDatabaseFilterButton();
+      await tester.tapCreateFilterByFieldType(
+        FieldType.Checkbox,
+        'Registration Complete',
+      );
```

Comment:
Failure caused by incorrect refresh of the frontend, leading to the new row being inserted at the wrong index. Refreshing the database view shows the correct row order.